homeassistant/components/yeelight/config_flow.py

```diff
@@ -38,22 +42,68 @@ def async_get_options_flow(config_entry):
     def __init__(self):
         """Initialize the config flow."""
         self._discovered_devices = {}
+        self._discovered_model = None
+        self._discovered_ip = None
+
+    async def async_step_homekit(self, discovery_info):
+        """Handle discovery from homekit."""
+        self._discovered_ip = discovery_info["host"]
+        return await self._async_handle_discovery()
+
+    async def async_step_dhcp(self, discovery_info):
+        """Handle discovery from dhcp."""
+        self._discovered_ip = discovery_info[IP_ADDRESS]
+        return await self._async_handle_discovery()
+
+    async def _async_handle_discovery(self):
+        """Handle any discovery."""
+        self.context[CONF_HOST] = self._discovered_ip
+        for progress in self._async_in_progress():
+            if progress.get("context", {}).get(CONF_HOST) == self._discovered_ip:
+                return self.async_abort(reason="already_in_progress")
+
+        self._discovered_model = await self._async_try_connect(self._discovered_ip)
+        if not self.unique_id:
+            return self.async_abort(reason="cannot_connect")
+
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: self._discovered_ip}, reload_on_update=False
```

Comment:
Why not reload?

Reply:
The reload listener will trigger a reload as well.